workers/unity/Packages/com.improbable.gdk.core/Systems/CommandBookkeepingSystem.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Improbable.Gdk.Core.Commands;
+using Unity.Entities;
+
+namespace Improbable.Gdk.Core
+{
+    public class CommandBookkeepingSystem : ComponentSystem
+    {
+        private Dictionary<Type, CommandStorage> commandStorageInstances = new Dictionary<Type, CommandStorage>();
+
+        public TConcreteCommandStorage GetCommandStorageForType<TConcreteCommandStorage>()
+            where TConcreteCommandStorage : CommandStorage
+        {
+            if (!commandStorageInstances.TryGetValue(typeof(TConcreteCommandStorage), out var storage))
+            {
+                throw new ArgumentException(
+                    $"Could not find command storage of type {nameof(TConcreteCommandStorage)}");
```

Comment:
I was told a nameof on a generic type just becomes the generic. @jessicafalk

Reply:
yes, that is true